sources/packages/api/src/route-controllers/application/application.student.controller.ts

```diff
@@ -112,12 +122,54 @@ export class ApplicationController extends BaseController {
         `Application id ${applicationId} was not found.`,
       );
     }
+    // Check wether the selected location is designated or not.
+    // If selected location is not designated, then make the
+    // selectedLocation null
+    if (application.data?.selectedLocation) {
+      const designatedLocation =
+        await this.locationService.getDesignatedLocationById(
+          application.data.selectedLocation,
+        );
+      const selectedLocation = await this.locationService.getLocationById(
+        application.data.selectedLocation,
+      );
+      if (!designatedLocation) application.data.selectedLocation = null;
+      // Assign location name for readonly form
+      application.data.selectedLocationName = selectedLocation?.name;
+    }
+    // Check wether the program is approved or not.
+    // If selected program is not approved, then make the
+    // selectedLocation null
+    if (application.data?.selectedProgram) {
+      const selectedProgram = await this.programService.getProgramById(
+        application.data.selectedProgram,
+      );
+
+      if (selectedProgram) {
+        // Assign program name for readonly form
+        application.data.selectedProgramName = selectedProgram.name;
+        if (selectedProgram.approvalStatus !== ApprovalStatus.approved)
```

Comment:
I believe that we have mostly added the curly braces even for single instructions.